app/stac_api/views_mixins.py

```diff
@@ -86,7 +86,14 @@ def update(self, request, *args, **kwargs):
     @transaction.atomic
     def upsert(self, request, *args, **kwargs):
         data = self.get_write_request_data(request, *args, **kwargs)
-        serializer = self.get_serializer(data=data)
+        if self.check_existence():
+            instance = self.get_object()
+            partial = kwargs.pop('partial', False)
+            serializer_kwargs = {'partial': partial}
+            serializer = self.get_serializer(instance, data=data, **serializer_kwargs)
+        else:
+            serializer = self.get_serializer(data=data)
```

Comment:
That's what we need to avoid here by using the django update_or_create() method. This code above don't treat race condition, and thats probably why the unittest are failing. Moreover the `check_existence()` is not correct as it returns a filter (`exists()` missing). To achieve our goal we don't want to provide the instance to the serializer.

Reply:
I see, I'll revert the check_existence(). And yes, I forgot to add the `.exists()`, thanks. But the unit tests were failing also before, that's why I re-added the `check_existence()` and also passed the instance to the serializer. Without this, the tests sometimes fail and sometimes pass. But also with this. So there must be something else. I am on it :-)